src/dpdkstat.c

```diff
@@ -0,0 +1,775 @@
+    g_configuration->eal_initialized = 0;
+    ssnprintf(g_configuration->coremask, DATA_MAX_NAME_LEN, "%s", "0xf");
+    ssnprintf(g_configuration->memory_channels, DATA_MAX_NAME_LEN, "%s", "1");
+    ssnprintf(g_configuration->process_type, DATA_MAX_NAME_LEN, "%s", "secondary");
+    ssnprintf(g_configuration->file_prefix, DATA_MAX_NAME_LEN, "%s",
+             "/var/run/.rte_config");
+
+    for (int i = 0; i < RTE_MAX_ETHPORTS; i++)
+      g_configuration->port_name[i][0] = 0;
+}
+
+static int dpdk_config(oconfig_item_t *ci)
+{
+  int port_counter = 0;
+
+  /* Initialize a POSIX SHared Memory (SHM) object. */
+  int err = dpdk_shm_init(sizeof(dpdk_config_t));
+  if (err) {
+    DEBUG("dpdkstat: error in shm_init, %s", strerror(errno));
+    return -1;
+  }
+
+  /* Set defaults for config, overwritten by loop if config item exists */
+  dpdk_config_init_default();
+
+  for (int i = 0; i < ci->children_num; i++) {
+    oconfig_item_t *child = ci->children + i;
+
+    if (strcasecmp("Interval", child->key) == 0) {
+      g_configuration->interval =
+            DOUBLE_TO_CDTIME_T (child->values[0].value.number);
+      DEBUG("dpdkstat: Plugin Read Interval %lu milliseconds\n",
+            CDTIME_T_TO_MS(g_configuration->interval));
+    } else if (strcasecmp("Coremask", child->key) == 0) {
+      ssnprintf(g_configuration->coremask, DATA_MAX_NAME_LEN, "%s",
+               child->values[0].value.string);
+      DEBUG("dpdkstat:COREMASK %s \n", g_configuration->coremask);
+      g_configuration->eal_argc+=1;
+    } else if (strcasecmp("MemoryChannels", child->key) == 0) {
+      ssnprintf(g_configuration->memory_channels, DATA_MAX_NAME_LEN, "%s",
+               child->values[0].value.string);
+      DEBUG("dpdkstat:Memory Channels %s \n", g_configuration->memory_channels);
+      g_configuration->eal_argc+=1;
+    } else if (strcasecmp("SocketMemory", child->key) == 0) {
+      ssnprintf(g_configuration->socket_memory, DATA_MAX_NAME_LEN, "%s",
+               child->values[0].value.string);
+      DEBUG("dpdkstat: socket mem %s \n", g_configuration->socket_memory);
+      g_configuration->eal_argc+=1;
+    } else if (strcasecmp("ProcessType", child->key) == 0) {
+      ssnprintf(g_configuration->process_type, DATA_MAX_NAME_LEN, "%s",
+               child->values[0].value.string);
+      DEBUG("dpdkstat: proc type %s \n", g_configuration->process_type);
+      g_configuration->eal_argc+=1;
+    } else if (strcasecmp("FilePrefix", child->key) == 0) {
+      ssnprintf(g_configuration->file_prefix, DATA_MAX_NAME_LEN, "/var/run/.%s_config",
+               child->values[0].value.string);
+      DEBUG("dpdkstat: file prefix %s \n", g_configuration->file_prefix);
+      if (strcasecmp(g_configuration->file_prefix, "/var/run/.rte_config") != 0) {
+        g_configuration->eal_argc+=1;
+      }
+    } else if (strcasecmp("EnabledPortMask", child->key) == 0) {
+      g_configuration->enabled_port_mask = (uint32_t)child->values[0].value.number;
```

Comment:
Please check that the config value is of the right type before accessing the union:

```c
if (child->values[0].type == OCONFIG_TYPE_NUMBER)
  …
```